test/main.js

```diff
@@ -62,38 +50,45 @@ describe('rebuildNativeModules', function() {
 
   const moduleVersionsToTest = ['0.22.0', '0.31.2'];
 
-  for(let nativeModuleVersionToBuildAgainst of moduleVersionsToTest) {
-    it(`Rebuilds native modules against ${nativeModuleVersionToBuildAgainst}`, async () => {
-      const targetHeaderDir = path.join(__dirname, 'testheaders');
-      const targetModulesDir = path.join(__dirname, 'node_modules');
-
+  for(let version of moduleVersionsToTest) {
+    it(`Rebuilds native modules against ${version}`, async () => {
       try {
-        if (await fs.stat(targetHeaderDir)) {
-          await rimraf(targetHeaderDir);
-        }
-      } catch (e) { }
-
-      await fs.mkdir(targetHeaderDir);
-
-      try {
-        if (await fs.stat(targetModulesDir)) {
-          await rimraf(targetModulesDir);
-        }
-      } catch (e) { }
-
-      await installNodeHeaders(nativeModuleVersionToBuildAgainst, null, targetHeaderDir);
-      let canary = await fs.stat(path.join(targetHeaderDir, '.node-gyp', nativeModuleVersionToBuildAgainst, 'common.gypi'));
-      expect(canary).to.be.ok
-
-      // Copy our own node_modules folder to a fixture so we don't trash it
-      await cp(path.resolve(__dirname, '..', 'node_modules'), targetModulesDir);
```

Comment:
> I actually don't mind slow tests,

I, respectfully, **very much mind** slow tests

> node_modules folder is a realistic test

i've proposed using a mocked module, which npm thinks is a _real_ module, that has actions post-build.  i then go in and assert that the build has completed, by testing that files exist that _would only exist if the build ran_.  I believe it to be very realistic, and much faster than building a set of modules from our src